File: frontend/src/shared/types/employee.ts
/**
 * 统一的Employee类型 - 所有模块都使用这个
 */
export interface Employee {
    id: string;
    name: string;
    description: string;
    avatar: string;
    category: string[];
    tags: string[];
    price: number | 'free';
    originalPrice?: number;
    trialCount: number;
    hireCount: number;
    isHired: boolean;
    isRecruited: boolean;
    isInTrial?: boolean;
    hiredAt?: string;
    createdAt?: string;
    createdBy?: string;
    status?: 'published' | 'archived'|'active' | 'inactive' | 'draft';
    skills?: string[];
    knowledgeBaseIds?: string[];
    isHot?: boolean;
    
    // 数字员工特有字段（可选）
    industry?: string;
    role?: string;
    prompt?: string;
    model?: string;
  }
  
  /**
   * 安全的数据规范化函数
   */
  export const normalizeEmployee = (data: any): Employee => {
    const name = data?.name || '未命名员工';
    const tags = Array.isArray(data?.tags) ? data.tags : [];
    
    // 检查是否为创建的数字员工
    const isCreated = tags.includes('created');
    
    return {
      id: data?.id || `emp-${Date.now()}`,
      name,
      description: data?.description || '',
      avatar: data?.avatar?.trim() ? data.avatar : `https://ui-avatars.com/api/?name=${encodeURIComponent(name)}&background=random`,
      category: Array.isArray(data?.category) ? data.category : 
                typeof data?.category === 'string' ? [data.category] : 
                isCreated ? ['created'] : ['默认'],
      tags,
      price: typeof data?.price === 'number' ? data.price : 
             data?.price === 'free' ? 'free' : 
             typeof data?.price === 'string' ? parseFloat(data.price) || 0 : 0,
      originalPrice: data?.originalPrice,
      trialCount: data?.trialCount || 0,
      hireCount: data?.hireCount || 0,
      isHired: Boolean(data?.isHired),
      isRecruited: Boolean(data?.isRecruited),
      isInTrial: data?.isInTrial,
      hiredAt: data?.hiredAt,
      createdAt: data?.createdAt || new Date().toISOString(),
      createdBy: data?.createdBy,
      status: data?.status,
      skills: data?.skills,
      knowledgeBaseIds: data?.knowledgeBaseIds,
      isHot: data?.isHot,
      
      // 数字员工特有字段
      industry: data?.industry,
      role: data?.role,
      prompt: data?.prompt,
      model: data?.model,
    };
  };
  
  /**
   * 创建安全的默认员工
   */
  export const createSafeEmployee = (overrides?: Partial<Employee>): Employee => ({
    id: 'unknown',
    name: '未知员工',
    description: '',
    avatar: 'https://ui-avatars.com/api/?name=Unknown&background=random',
    category: ['默认'],
    tags: [],
    price: 0,
    trialCount: 0,
    hireCount: 0,
    isHired: false,
    isRecruited: false,
    ...overrides
  });

File: backend-python-ai/app/db/models/employee.py
"""
员工ORM模型
对应SQL: employees
"""

from datetime import datetime
from typing import Optional, List

from sqlalchemy import Column, String, DateTime, Boolean, Text, Integer, ForeignKey, JSON
from sqlalchemy.orm import relationship

from app.db.database import Base
from app.db.models.base import TimestampMixin


class Employee(Base, TimestampMixin):
    """数字员工表"""
    
    __tablename__ = "employees"
    
    id = Column(String(50), primary_key=True)
    name = Column(String(100), nullable=False)
    description = Column(Text)
    avatar = Column(Text)
    category = Column(JSON, default=list)
    tags = Column(JSON, default=list)
    price = Column(String(20), default="0")
    original_price = Column(Integer)
    trial_count = Column(Integer, default=0)
    hire_count = Column(Integer, default=0)
    is_hired = Column(Boolean, default=False)
    is_recruited = Column(Boolean, default=False)
    status = Column(String(20), default="draft")
    skills = Column(JSON, default=list)
    knowledge_base_ids = Column(JSON, default=list)
    industry = Column(String(50))
    role = Column(String(50))
    prompt = Column(Text)
    model = Column(String(50), default="deepseek-chat")
    model_config = Column(JSON, default=dict)
    is_hot = Column(Boolean, default=False)
    personality = Column(String(500))
    welcome_message = Column(Text)
    created_by = Column(String(50), ForeignKey("users.id"))
    organization_id = Column(String(50), ForeignKey("organizations.id"))
    
    # 员工类型：system-系统级（管理员创建），user-用户级（普通用户创建）
    employee_type = Column(String(20), default="user", index=True)
    # 是否为系统默认员工
    is_system_default = Column(Boolean, default=False, index=True)
    
    # 关联关系
    creator = relationship("User", foreign_keys=[created_by], back_populates="created_employees")
    organization = relationship("Organization", back_populates="employees")
    conversations = relationship("Conversation", back_populates="employee")
    hire_records = relationship("HireRecord", back_populates="employee")
    trial_records = relationship("TrialRecord", back_populates="employee")
